rust/worker/src/assignment/rendezvous_hash.rs

```diff
@@ -0,0 +1,173 @@
+// This implementation mirrors the rendezvous hash implementation
+// in the go and python services.
+// The go implementation is located go/internal/utils/rendezvous_hash.go
+// The python implementation is located chromadb/utils/rendezvous_hash.py
+
+use crate::errors::{ChromaError, ErrorCodes};
+use std::io::Cursor;
+use thiserror::Error;
+
+use murmur3::murmur3_x64_128;
+
+/// A trait for hashing a member and a key to a score.
+trait Hasher {
+    fn hash(&self, member: &str, key: &str) -> Result<u64, AssignmentError>;
+}
+
+/// Error codes for assignment
+#[derive(Error, Debug)]
+enum AssignmentError {
+    #[error("Cannot assign empty key")]
+    EmptyKey,
+    #[error("No members to assign to")]
+    NoMembers,
+    #[error("Error hashing member")]
+    HashError,
+}
+
+impl ChromaError for AssignmentError {
```

Comment:
We must always do this, it's not compile time enforced, I can't find a good way to do that right now, but higher levels of code can Box<dyn ChromaError> this way and rely on codes.